### PR TITLE
docs: improve description of committing in contributing

### PR DESCRIPTION
**Pull request by haved**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -14,9 +14,11 @@ The commit messages are used to automatically bump the versions of changed packa
 
 We use semantic versioning, with versions `MAJOR.MINOR.PATCH`.
 
- - For fixes, start the commit with `fix:`. This will bump PATCH.
- - For features, start the commit with `feat:`. This will bump MINOR.
- - For changes that break backwards compatability, add a footer to the commit description.
+ - For fixes, start the commit with `fix: `. This will bump PATCH.
+ - For features, start the commit with `feat: `. This will bump MINOR.
+ - For changes that break backwards compatability, add a `!` before the colon: `feat!: `.
```
Comment (polomani) [inline CONTRIBUTING.md]:
```suggestion
 - For changes that break backwards compatibility, add a `!` before the colon: `feat!: `.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -14,9 +14,11 @@ The commit messages are used to automatically bump the versions of changed packa
...
+   
+   Alternativly you may specify a footer (or both).
    The footer must come after a blank line, and start with `BREAKING CHANGE:` (colon mandatory),
    followed by a description of what breaks.
  - For extra details in the changelog, you can specify a scope, e.g. `feat(assets):`.
```
Comment (polomani) [inline CONTRIBUTING.md]:
```
For other changes, use `refactor:`, `test:`, `docs:`, `chore:` etc.
Try something out, and the commit hook will tell you what's allowed.
```
maybe we can explain more here?
like chore is for changes related to project setup, etc

and provide a full list of options

**Comment by haved** [comment]
This PR now also contains changes to `commitlint.config.js`, where a custom parser is used to parse the header. This now allows for `!` specifiers in headers. In addition I tried to make it harder to accidentally misformat a `BREAKING CHANGE: ` footer by requiring a blank line before the footer (which is required for MAJOR bump).

Note: commitlint has extra rules built in to ignore (not limited to):
 - `Merge pull request *`
 - `Merge branch *`
 - `Revert *`
 - `revert *`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -2,21 +2,43 @@ Contributing
...
+   - `ci: ` changes to ci scripts and pipeline
+   - `refactor: ` code moving and renaming
+   - `style: ` fixes to code style
+   - `perf: ` changes to improve performance
+   - `revert: ` changing things back 
+   - `chore: ` miscelanious changes
+
+#### Example
+```
+docs: add example of commit with subject line to CHANGELOG
```
Comment (polomani) [inline CONTRIBUTING.md]:
```suggestion
docs(changelog): add example of commit with subject line
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -2,21 +2,43 @@ Contributing
...
+   - `ci: ` changes to ci scripts and pipeline
+   - `refactor: ` code moving and renaming
+   - `style: ` fixes to code style
+   - `perf: ` changes to improve performance
+   - `revert: ` changing things back 
+   - `chore: ` miscelanious changes
+
+#### Example
+```
+docs(changelog): add example of commit with subject line
```
Comment (BugGambit) [inline CONTRIBUTING.md]: Would you ever have "changelog" here?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -35,7 +35,7 @@ We use semantic versioning, with versions `MAJOR.MINOR.PATCH`.
 
 #### Example
 ```
-docs: add example of commit with subject line to CHANGELOG
+docs(changelog): add example of commit with subject line
```
Comment (polomani) [inline CONTRIBUTING.md]:
```suggestion
docs(contributing-readme): add example of commit with subject line
```

**Pull request merged**